src/vmm/src/guest_config/templates/mod.rs

```diff
@@ -119,32 +283,212 @@ pub mod x86_64 {
 
-/// Guest config sub-module specifically useful for
-/// CPU templates to configure aarch64 CPUs.
-#[cfg(target_arch = "aarch64")]
-pub mod aarch64 {}
+    /// CPU template with modifiers to be applied on
+    /// top of an existing configuration to generate
+    /// the guest configuration to be used.
+    pub fn create_guest_cpu_config(
+        _template: &Aarch64CpuTemplate,
+        _host_config: &Aarch64CpuConfiguration,
+    ) -> Result<Aarch64CpuConfiguration, Error> {
+        // TODO
+        Ok(Aarch64CpuConfiguration::default())
+    }
+
+    fn deserialize_u128_from_str<'de, D>(deserializer: D) -> Result<u128, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let number_str = String::deserialize(deserializer)?;
+        let deserialized_number: u128 = if number_str.len() > 2 {
+            match &number_str[0..2] {
+                "0b" => u128::from_str_radix(&number_str[2..], 2),
+                "0x" => u128::from_str_radix(&number_str[2..], 16),
+                _ => u128::from_str(&number_str),
+            }
+            .map_err(|err| {
+                D::Error::custom(format!(
+                    "Failed to parse string [{}] as a number for CPU template - {:?}",
+                    number_str, err
+                ))
+            })?
+        } else {
+            u128::from_str(&number_str).map_err(|err| {
+                D::Error::custom(format!(
+                    "Failed to parse string [{}] as a decimal number for CPU template - {:?}",
+                    number_str, err
+                ))
+            })?
+        };
+
+        Ok(deserialized_number)
+    }
+}
 
 #[cfg(test)]
 mod tests {
     #[cfg(target_arch = "x86_64")]
-    use crate::guest_config::templates::x86_64::X86_64CpuTemplate;
+    use kvm_bindings::KVM_CPUID_FLAG_STATEFUL_FUNC;
+    #[cfg(target_arch = "x86_64")]
+    use kvm_bindings::KVM_MAX_CPUID_ENTRIES;
+    #[cfg(target_arch = "x86_64")]
+    use kvm_ioctls::Kvm;
+
+    #[cfg(target_arch = "x86_64")]
+    use crate::guest_config::cpuid::KvmCpuidFlags;
+    #[cfg(target_arch = "x86_64")]
+    use crate::guest_config::cpuid::{Cpuid, RawCpuid};
+    #[cfg(target_arch = "aarch64")]
+    use crate::guest_config::templates::aarch64::{Aarch64CpuTemplate, BitMask, RegisterModifier};
+    #[cfg(target_arch = "x86_64")]
+    use crate::guest_config::templates::x86_64::{
+        create_guest_cpu_config, BitMask, CpuidLeafModifier, CpuidRegister, CpuidRegisterModifier,
+        RegisterModifier, X86_64CpuTemplate,
+    };
+    #[cfg(target_arch = "x86_64")]
+    use crate::guest_config::x86_64::X86_64CpuConfiguration;
 
     #[cfg(target_arch = "x86_64")]
     const X86_64_TEMPLATE_JSON: &str = r#"{
+        "cpuid_modifiers": [
+            {
+                "leaf": "0x80000001",
+                "subleaf": "0x0007",
+                "flags": 0,
+                "modifiers": [
+                    {
+                        "register": "eax",
+                        "mask": {
```

Comment:
I see now what caused my confusion last time. Do we need this intermediate change that is removed in a later commit?